docs/plugin-hierarchy.md

```diff
@@ -0,0 +1,28 @@
+## Plugin hierarchy
+
```

Comment:
Can we include here the information which plugins are public and which internal? Or maybe also which are external (not owned by Shipkit)